### ISSUE TITLE
torch.finfo(dtype).min is 0 when dtype is torch.float16 while in pytorch the min value is -65504.0

### ISSUE THREAD

**Issue by LittleLittleCloud** [dotnet/TorchSharp#1279]
torchsharp version: 0.101.5

**Comment by yueyinqiu** [comment]
It seems that someone forgot to deal with float16. I will create a pr soon. However it may not be merged and published in a short time for some reason.

**Comment by yueyinqiu** [comment]
Ahh great... That's happening in my PyTorch:

```python
dtype = torch.float16
tensor = torch.tensor(1, dtype=dtype)
eps = torch.tensor(torch.finfo(dtype).eps * 0.9, dtype=dtype)
print(tensor == tensor + eps) // False
```

Why `tensor + eps` is not equals to `tensor`... I'm now not sure about that...

**Comment by yueyinqiu** [comment]
I've found that the result of other scalar types are not correct as well (at least not same as PyTorch). I will try to fix this at the same time.

**Comment by NiklasGustafsson** [comment]
Just released v0.102.4, which has a fix for this.
